Added RabbitMQ event_source

ttestscripting opened the pull request:
This PR adds support for RabbitMQ as an event source and includes an example rulebook. It only supports user/pass auth in its present state.

bzwei commented on plugins/event_source/rabbitmq.py:
```diff
@@ -0,0 +1,75 @@
+            async for message in queue_iter:
+                async with message.process():
+                    logger.info(message.body.decode())
+                    data = dict(
```
We want all plugins to generate events in similar format. Please take a look at kafka.py as an example. The message body itself must be a json str and is converted into a dict. No need to use meta to store the information about the message bus itself for every event.

ttestscripting replied:
Converted the format as requested

mkot02 commented:
@ttestscripting are you going to fix this PR to merge it into the master branch? I'm interested in using RabbitMQ as event source, and I'm not sure if I should take over your work and fix issues indicated here or maybe you're already working on this.

ttestscripting commented:
[mkot02](https://github.com/mkot02) [bzwei](https://github.com/bzwei) Sorry for the late responses, I just got the chance to review this. I added SSL support and converted the code to allow for the JSON changes as requested. An example rule-book for SSL support is below as well normal username and password auth. Let me know if anything else is needed. 

````
---
- name: Listen for events on a rabbitmq
  hosts: all
  sources:
    - ansible.eda.rabbitmq:
        host: localhost
        port: 5672
        queue: hello
        username: test
        password: test
    - ansible.eda.rabbitmq:
        host: localhost
        port: 5671
        username: test
        password: test
        queue: hello
        cafile: /certificates/ca_certificate.pem
        certfile: /certificates/client_certificate.pem
        keyfile: /certificates/client_key.pem
        check_hostname: false
        certpassword: secretpass
  rules:
    - name: queue is hello and message body is world
      condition: event.name == "world"
      action:
        debug:
````

Alex-Izquierdo commented:
Hi @ttestscripting, the PR has conflicts to resolve.

Alex-Izquierdo commented:
Hi @ttestscripting 

We want to express our sincere appreciation for your effort and contribution. After careful consideration, we have concluded that this plugin doesn't align with the scope of our official collection, and unfortunately, it will not be included.

However, this does not diminish the value of your work. We believe your plugin can still be highly beneficial and encourage you to offer it through your own collection. You are welcome to upload it [to galaxy](https://galaxy.ansible.com/) with the "eda" tag.

Thank you for your understanding and contribution.

miralgj commented:
@ttestscripting, did you ever publish this on galaxy? If not, would it be alright with you if I took your contribution and published it on galaxy and maintained it?

ttestscripting commented:
@miralgj I haven't published this on Galaxy yet, so feel free to take it and share it there.
